get ticket scanning nominally working

chas-mcmahon opened the pull request:
connect #238
@daybreaker This is ready for review. I left a lot of questions/comments and todos in the code, let me know if you know the answers to any of 'em or if they don't make sense.

~~WIP / don't merge~~

~~needs some design help before merging, but automatic scanning seems to be working. I'm not sure if there need to be more steps for the failure modes, but I think I saw a comment on the manual check in ticket about alternatives to scanning in case of server errors, device malfunctions, or w/e else.~~

~~adding @daybreaker for review since the technical bits are ready enough for another pair of eyes~~

daybreaker commented on src/Accounts/eventManager.js:
```diff
@@ -9,17 +10,58 @@ const styles = SharedStyles.createStyles()
 const eventManagerStyles = EventManagerStyles.createStyles()
 
-
+// TODO: probably should make this a class component
```
Yes please.

Dont be shy about converting things to components. I usually only make them functions at first when creating simple landing pages for Brittany. If they have to interact with the store, or have too many internal methods, I say convert to Component.

daybreaker commented on src/Accounts/eventManager.js:
```diff
@@ -9,17 +10,58 @@ const styles = SharedStyles.createStyles()
+  // presumably use stale tickets that they never redeemed to enter new events
+  const scanForEvent = (event) => {
+    return () => {
+      eventManager.setState({eventToScan: event}, () => {navigate('EventScanner')});
```
I dont know if I like setting state directly from EventManager. I think the best practice for Unstated is to create a function in the EventManagerStateProvider like `scanEvent = async (event) => { /* setState */ } `

then in here, you can do

```
await eventManager.scanEvent(event)
navigate('EventScanner')
```

daybreaker commented on src/Accounts/eventManager.js:
```diff
@@ -9,17 +10,58 @@ const styles = SharedStyles.createStyles()
+          <View style={eventManagerStyles.cardImageWrapper}>
+            <Image
+              style={eventManagerStyles.cardImage}
+              source={require('../../assets/doorman-event-img-3.png')}
```
We should have images from the event itself to not need placeholders.

daybreaker commented on src/Accounts/eventManager.js:
```diff
@@ -9,17 +10,58 @@ const styles = SharedStyles.createStyles()
+
+    eventManager.getEvents();
+  }
+  // TODO: not sure if the NavigationEvents thing is the right way to get at this data
```
Yeah, its fine.

The way the nav stack works is it will only load the component once. So even when you go backwards in the nav stack, the nav to this page again, it wont re-fire the constructor or componentWill/DidMount events, because its still technically mounted.

onWillFocus is sort of like onWillMount except it fires every time you navigate to this screen.

Grabbing events here means the user will always have the most recent events. So it's good.

daybreaker commented on src/Accounts/eventManager.js:
```diff
@@ -9,17 +10,58 @@ const styles = SharedStyles.createStyles()
+        {eventManager.events.map(EventCard)}
 
-        <TouchableHighlight underlayColor="rgba(0, 0, 0, 0)" onPress={() => navigate('EventScanner')}>
+        {/* <TouchableHighlight underlayColor="rgba(0, 0, 0, 0)" onPress={() => navigate('EventScanner')}>
```
You can just delete any of these leftover mock ups.

daybreaker commented on src/state/eventManagerStateProvider.js:
```diff
@@ -12,9 +13,27 @@ class EventManagerContainer extends Container {
+    return this.state.events
+  }
+
+  // TODO: filter by live vs upcoming?
```
So, I think there's a different way to get these events, rather than the normal events list.

Based on the bn-web app, it looks like a user has organizations, and an organization can have events. But it also looks like it's not fully implemented yet?

https://github.com/big-neon/bn-web/blob/master/src/components/pages/admin/events/List.js#L84

You might want to ask Keith which endpoint is the best for getting the Doorman EventManager event listing, or if for now maybe the normal event listing is fine?

chas-mcmahon commented:
@daybreaker I addressed most of the feedback and it's mostly g2g, but for some reason the event images aren't loading. The urls seem fine, though I wasn't able to check them independently before I had to put this down. Not sure what's wrong there.

I also didn't have time to look into event filtering.
